src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/ReflectMethods.java

```diff
@@ -2679,7 +2679,7 @@ FunctionType typeToFunctionType(Type t) {
         }
 
         RecordTypeRef symbolToRecordTypeRef(Symbol.ClassSymbol s) {
-            TypeElement recordType = typeToTypeElement(s.type);
+            TypeElement recordType = typeToTypeElement(s.erasure(types));
```

Comment:
The new grammar for record references wants an erased type identifier here, so I tweaked the code accordingly.